graphlearn/src/core/graph/storage/types.h

```diff
@@ -32,39 +32,160 @@ typedef std::vector<IndexType> IndexList;
 typedef std::vector<std::vector<IdType>> IdMatrix;
 typedef std::unordered_map<IdType, IndexType> MAP;
 
+template <class T>
+class MultiArray {
+public:
+  MultiArray(const std::vector<const T*>& values, const std::vector<int32_t> sizes,
```

Comment:
What is the purpose of using MultiArray and RangeArray?

Reply:
The `MultiArray` is a view that treats multiple chunked array as a consective array without extra copying.

The `RangeArray` is a view that treats a range as a array without actually generating such an array to save memory.

Reply:
It appears that the current code does not use these classes.